data-prepper-plugins/grok-prepper/README.md

```diff
@@ -1,59 +1,241 @@
 # Grok Prepper
 
 This is a prepper that takes unstructured data and utilizes pattern matching
-to structure and extract important fields for easier and more insightful aggregation and analysis.
+to structure and extract important keys and make data more structured and queryable.
 
-## Usages
+The Grok Prepper uses the [java-grok Library](https://github.com/thekrakken/java-grok) internally and supports all java-grok library compatible patterns. The java-grok library is built using the `java.util.regex` regular expression library.
+
+The full set of default patterns can be found [here](https://github.com/thekrakken/java-grok/blob/master/src/main/resources/patterns/patterns). Custom patterns can be added through either the
+`patterns_definitions` or `patterns_directories` configuration settings. When debugging custom patterns, the [Grok Debugger](https://grokdebug.herokuapp.com/) 
+can be extremely helpful.
+
+## Basic Grok Usage
+
+To get started with grok using Data Prepper, create the following `pipeline.yaml`.
+```yaml
+grok-pipeline:
+  source:
+    file:
+      path: "/full/path/to/grok_logs_json.log"
+  prepper:
+    - grok:
+        match:
+          message: ['%{IPORHOST:clientip} \[%{HTTPDATE:timestamp}\] %{NUMBER:response_status:int}']
+  sink:
+    - stdout:
+```
+
+Create a file named `grok_logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with the path of this file.
+
+The grok configuration from the `pipeline.yaml` will match the value in the `message` key of each log for a pattern matching `%{IPORHOST:clientip} \[%{HTTPDATE:timestamp}\] %{NUMBER:response_status:int}`.
+These three patterns (`IPORHOST`, `HTTPDATE`, and `NUMBER`) are default patterns.
 
-Example `.yaml` configuration
+Copy the following log into `grok_logs_json.log` in order to align with this.
```

Comment:
Nit and you can leave this unchanged too, maybe make this a part of line 28 like:
Create the following file named `grok_logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with the path of this file.
```
{"message": "127.0.0.1 198.126.12 [10/Oct/2000:13:55:36 -0700] 200"}
```